Src/FluentAssertions/Xml/XmlReaderValidator.cs

```diff
@@ -77,7 +101,15 @@ private ValidationResult Validate()
                             return validationResult;
                         }
 
+                        // starting new element, add local name to location stack
+                        // to build XPath info
                         locationStack.Push(subjectReader.LocalName);
+
+                        // building key according XPath to count repeating element names
+                        var locationKey = "/" + string.Join("/", locationStack.Reverse());
+                        locationCounts.TryGetValue(locationKey, out var locationCount);
+                        locationCounts[locationKey] = ++locationCount;
```

Comment:
🔧How about extracting these lines into a method with some appropriate method name?

Reply:
Mmh, this is the only usage, but it reduces the LOC in the `switch`.
Maybe, more statements should be moved to method (in another PR)!?